app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -88,7 +88,9 @@ class CheckAgencyController @Inject()
                                             (implicit authContext: AuthContext, request: Request[AnyContent]): Future[Result] = {
     agentSubscriptionConnector.getRegistration(knownFacts.utr, knownFacts.postcode) map { maybeRegistration: Option[Registration] =>
       maybeRegistration match {
-        case Some(_) => Redirect(routes.CheckAgencyController.showConfirmYourAgency())
+        case Some(Registration(Some(name))) => Redirect(routes.CheckAgencyController.showConfirmYourAgency())
+                          .flashing("agencyName" -> name, "knownFactsPostcode" -> knownFacts.postcode, "utr" -> knownFacts.utr)
```

Comment:
The story says `Please use the Name and postcode data from this api
https://confluence.tools.tax.service.gov.uk/display/AG/Registration` but this gets the postcode from the user input, not the API.

I think that the code is probably OK as it is, the only possible difference is the casing and spacing of the postcode displayed, but please can you confirm with JP and assuming he's OK with it update JIRA to match the code?